src/Tools/dotnet-trace/DiagnosticProfileHandlers/LoaderBinderHandler.cs

```diff
@@ -0,0 +1,154 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+using System;
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Tools.Trace.DiagnosticProfileHandlers
+{
+    /// <summary>
+    /// This class handles parsing for loader and binder CLR events.
+    /// </summary>
+    internal class LoaderBinderHandler : IDiagnosticProfileHandler
+    {
+        public void AddHandler(EventPipeEventSource source)
+        {
+            source.Clr.AssemblyLoaderStart += Clr_AssemblyLoaderStart;
+            source.Clr.AssemblyLoaderStop += Clr_AssemblyLoaderStop;
+            source.Clr.AssemblyLoaderResolutionAttempted += Clr_AssemblyLoaderResolutionAttempted;
+            source.Clr.AssemblyLoaderAppDomainAssemblyResolveHandlerInvoked += Clr_AssemblyLoaderAppDomainAssemblyResolveHandlerInvoked;
+            source.Clr.AssemblyLoaderAssemblyLoadContextResolvingHandlerInvoked += Clr_AssemblyLoaderAssemblyLoadContextResolvingHandlerInvoked;
+            source.Clr.AssemblyLoaderAssemblyLoadFromResolveHandlerInvoked += Clr_AssemblyLoaderAssemblyLoadFromResolveHandlerInvoked;
+            source.Clr.AssemblyLoaderKnownPathProbed += Clr_AssemblyLoaderKnownPathProbed;
+            source.Clr.LoaderAssemblyLoad += Clr_LoaderAssemblyLoad;
+            source.Clr.LoaderAssemblyUnload += Clr_LoaderAssemblyUnload;
+            source.Clr.LoaderModuleLoad += Clr_LoaderModuleLoad;
+            source.Clr.LoaderModuleUnload += Clr_LoaderModuleUnload;
+        }
+
+        private void WriteLoaderBinderLog(string evtName, DateTime timestamp, string msg)
+        {
+            Console.WriteLine($"[CLR-LoaderBinder|{evtName}|{timestamp.ToString("s", CultureInfo.InvariantCulture)}] {msg}");
```

Comment:
Out of curiosity: are these the format that the current/previous loader-binder tools output?

More specifically, is there a reason we can't use the `CustomProviderHandler` logic for this?  e.g., existing analysis scripts that expect a specific output.

Reply:
There's no current/previous loader-binder tool afaik...? Unless you're talking about fuslogvw? 

You can look at what fuslogvw output looks like here:
https://docs.microsoft.com/en-us/dotnet/framework/tools/fuslogvw-exe-assembly-binding-log-viewer

The main reason I'm not using CustomProviderHandler is because some fields aren't very helpful (i.e. ClrInstanceID) or need string formatting (i.e. the assembly flags, assembly resolution result, etc.)